### ISSUE TITLE
PipeClient::read sometimes causes panic

### ISSUE THREAD

**Issue by ohomburg** [blackbeam/named_pipe#8]
The panic originates from `lib.rs:974:26` (`PipeClient::drop`) eventually tracing back to `lib.rs:1036` (the end of `ReadHandle::wait`). The message reads `CANCEL ERROR Os { code: 1168, kind: Other, message: "Element not found." }`.

It occurs when reading with a zero timeout, but only sometimes. My guess is a race condition.

<details><summary>Code sample</summary>

```rust
use std::io;
use std::io::{BufRead, BufReader, ErrorKind, Read};
use std::time::Duration;

fn main() -> io::Result<()> {
    let mut client = named_pipe::PipeClient::connect(r"\\.\pipe\abcd")?;
    client.set_read_timeout(Some(Duration::from_millis(0)));

    let mut reader = BufReader::new(client);
    loop {
        while let Some(line) = read_line(&mut reader)? {
            print!("read line: {}", line);
        }

        // do some unrelated work
    }
}

fn read_line<T: Read>(reader: &mut BufReader<T>) -> io::Result<Option<String>> {
    let mut line = String::new();
    match reader.read_line(&mut line) {
        Ok(_) => Ok(Some(line)),
        Err(ref err) if err.kind() == ErrorKind::TimedOut => Ok(None),
        Err(err) => Err(err),
    }
}
```
</details>

**Comment by blackbeam** [comment]
Hi and thanks for report. I'll look into it.

**Comment by blackbeam** [comment]
Fixed in [v0.4.1](https://github.com/blackbeam/named_pipe/releases/tag/v0.4.1)
